### PR TITLE
Move Media::AccountLogin cluster to match the spec

### PR DESCRIPTION
**Pull request by andy31415**
Changes:
   - mark the Login and GetSetupPin requiring administer privileges.
   - Add fabric-scoped flags to Login, Logout and GetSetupPin

Changes to be made in spec: 
  - remove "nullable" attribute to setup pin on reply for getting the setup pin

### CODE REVIEW

**Review by tcarmelveilleux — DISMISSED** [review]

**Review by yufengwangca — APPROVED** [review]

**Review by chrisdecenzo — APPROVED** [review]

File: examples/all-clusters-minimal-app/all-clusters-common/all-clusters-minimal-app.matter [hunk 1]
```diff
@@ -3645,11 +3645,11 @@ server cluster AccountLogin = 1294 {
   }
 
   response struct GetSetupPINResponse = 1 {
-    char_string setupPIN = 0;
+    nullable char_string setupPIN = 0;
```
Comment (andy31415) [inline examples/all-clusters-minimal-app/all-clusters-common/all-clusters-minimal-app.matter:3648]: Adding nullable seems to be a wire-backwards incompatibility.

Should we change spec instead? @bzbarsky-apple @chrisdecenzo

Comment (bzbarsky-apple) [reply examples/all-clusters-minimal-app/all-clusters-common/all-clusters-minimal-app.matter:3648]: Well, this is a wire-format-incompatible change, so... Probably change the spec at this point.

**Review by bzbarsky-apple — DISMISSED** [review]

File: examples/all-clusters-minimal-app/all-clusters-common/all-clusters-minimal-app.matter (hunk 1 above)
Comment (andy31415) [reply examples/all-clusters-minimal-app/all-clusters-common/all-clusters-minimal-app.matter]: Undid the setupPin nullable change. Will create a Spec PR instead for that. Changes in access remain.

Comment (andy31415) [reply examples/all-clusters-minimal-app/all-clusters-common/all-clusters-minimal-app.matter]: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7838 is the spec PR

File: examples/tv-app/tv-common/clusters/account-login/AccountLoginManager.cpp [hunk 2]
```diff
@@ -60,6 +60,6 @@ void AccountLoginManager::HandleGetSetupPin(CommandResponseHelper<GetSetupPINRes
     GetSetupPINResponse response;
     ChipLogProgress(Zcl, "temporary account id: %s returning pin: %s", tempAccountIdentifierString.c_str(), mSetupPin);
 
-    response.setupPIN = CharSpan::fromCharString(mSetupPin);
+    response.setupPIN = chip::app::DataModel::MakeNullable(CharSpan::fromCharString(mSetupPin));
```
Comment (bzbarsky-apple) [inline examples/tv-app/tv-common/clusters/account-login/AccountLoginManager.cpp:63]: This change should not be here, right? It does not compile.

File: examples/tv-app/android/java/ContentAppCommandDelegate.cpp [hunk 3]
```diff
@@ -321,11 +321,11 @@ GetSetupPINResponseType ContentAppCommandDelegate::FormatGetSetupPINResponse(Jso
         std::to_string(to_underlying(app::Clusters::AccountLogin::Commands::GetSetupPINResponse::Fields::kSetupPIN));
     if (!value[setupPINFieldId].empty())
     {
-        getSetupPINresponse.setupPIN.SetNonNull(CharSpan::fromCharString(value[setupPINFieldId].asCString()));
+        getSetupPINresponse.setupPIN = CharSpan::fromCharString(value[setupPINFieldId].asCString());
```
Comment (bzbarsky-apple) [inline examples/tv-app/android/java/ContentAppCommandDelegate.cpp:324]: Fwiw, pre-existing but I am pretty sure this is a use-after-free in general....

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/tv-app/android/java/ContentAppCommandDelegate.cpp (hunk 3 above)
Comment (andy31415) [reply examples/tv-app/android/java/ContentAppCommandDelegate.cpp:324]: created #30155 for this

**Pull request merged**